__std__/flake.nix

```diff
@@ -0,0 +1,249 @@
+{
+  description = "Plutus Core - The Scripting Language for Cardano";
+
+  # The flake inputs will be accessible by name in each nix file like so:
+  # { inputs, cell }: inputs.nixpkgs, inputs.haskell-nix
+  inputs = {
+    nixpkgs = {
+      url = "github:NixOS/nixpkgs/34e4df55664c24df350f59adba8c7a042dece61e";
+    };
+    std = {
+      url = "github:divnix/std";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    haskell-nix = {
+      url = "github:input-output-hk/haskell.nix";
+      inputs = {
+        hackage.follows = "hackage-nix";
+      };
+    };
+    hackage-nix = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    sphinxcontrib-haddock = {
+      url = "github:michaelpj/sphinxcontrib-haddock";
+      flake = false;
+    };
+    # cardano-repo-tool = {
+    #   url = "github:input-output-hk/cardano-repo-tool";
+    #   flake = false;
+    # };
+    # gitignore-nix = {
+    #   url = "github:hercules-ci/gitignore.nix";
+    #   flake = false;
+    # };
+    # haskell-language-server = {
+    #   # Pinned to a release
+    #   url = "github:haskell/haskell-language-server?ref=1.7.0.0";
+    #   flake = false;
+    # };
+    # iohk-nix = {
+    #   url = "github:input-output-hk/iohk-nix";
+    #   flake = false;
+    # };
+    # pre-commit-hooks-nix = {
+    #   url = "github:cachix/pre-commit-hooks.nix";
+    #   flake = false;
+    # };
+  };
+
+  # The flake outputs are managed by std.
+  outputs = inputs:
+
+    # The growOn function accepts a first argument defining the organelles.
+    inputs.std.growOn
+      {
+
+        # Boilerplate
+        inherit inputs;
+
+        # ALL nix files will reside inside this folder, no exception.
+        # Each subfolder is a "cell".
+        # Cell names are arbitrary.
+        # Cells are for highest-level organization and grouping of nix code.
+        #
+        # In this repository we have three cells:
+        #   doc
+        #     Develop and build all the documentation artifacts
+        #   haskell
+        #     Develop and build all haskell components
+        #   toolchain
+        #     Common tools and functions shared across multiple cells
+        cellsFrom = ./nix;
+
+        # Each cell contains arbitrary "organelles".
+        # Each organelle must be either:
+        #   A nix file named after the organelle
+        #   A directory named after the organelle and containing a default.nix
+        # Organelles have types.
+        # Not all cells have the same organelles.
+        # All organelles belong in a cell.
+        #
+        # In this repository we have five organelles, listed below with their type:
+        #   devshells :: devshells
+        #     Development shells available via nix develop
+        #   packages :: installables
+        #     Packages available via nix build
+        #   devshellProfiles :: functions
+        #     Building blocks for devshells, not exposed to the nix cli
```

Comment:
nix cli = nix-shell environment of the project?

Reply:
Mmm actually originally this was `to the nix flake`. 
But then @blaggacao pointed out that it's rather the `nix cli`.
Where `nix cli` is the nix command line interface itself, i.e. `nix (build|develop|run|...)`

However, I'm not sure anymore.
@blaggacao are you sure that `to the nix flake` was incorrect?
Indeed if I do not harvest the `devshellProfiles` organelle (which I don't), then it does not become part of the flake, correct?

Maybe you meant to say that `devshellProfiles` are not exposed to the nix cli, because there is nothing you can do with them? (indeed they are neither shells nor derivations, they are functions)